src/Xamarin.Android.Build.Tasks/Tasks/BuildApk.cs

```diff
@@ -258,6 +245,49 @@ void ExecuteWithAbi (string [] supportedAbis, string apkInputPath, string apkOut
 			}
 		}
 
+		private void UpdateEntriesFromInputApk (string apkInputPath, ZipArchiveEx apk, DateTime lastWriteOutput, DateTime lastWriteInput, HashSet<ulong> deletedEntries)
+		{
+			using (var packaged = new ZipArchiveEx (apkInputPath, FileMode.Open)) {
+				foreach (var entry in packaged.Archive) {
+					Log.LogDebugMessage ($"Deregistering item {entry.FullName}");
+					existingEntries.Remove (entry.FullName);
+					if (lastWriteInput <= lastWriteOutput)
+						continue;
+
+					long entryIndexInOutput;
+					if (apk.Archive.ContainsEntry (entry.FullName, out entryIndexInOutput)) {
+						ZipEntry e = apk.Archive.ReadEntry (entry.FullName);
+						// check the CRC values as the ModifiedDate is always 01/01/1980 in the aapt generated file.
+						if (entry.CRC == e.CRC) {
+							Log.LogDebugMessage ($"Skipping {entry.FullName} from {apkInputPath} as its up to date.");
+							continue;
+						}
+					} else {
+						entryIndexInOutput = -1;
+					}
+
+					var ms = new MemoryStream ();
+					entry.Extract (ms);
```

Comment:
Is this advisable or a good idea?

@grendello: is there a "good" libZipSharp method which can copy data *between* `.zip` files *without* decompressing them?

My fear is that `entry` may refer to e.g. a 2GB resource (`.mp4` file?), at which point this `entry.Extract(ms)` invocation is going to need to contain said 2GB of data, which very probably isn't going to happen, which will in turn cause "obscure" corner cases in `.zip` handling.

Maybe the "proper" thing to do is check `entry.Size` and use an intermediate file if it's "too big", though that has implications with Windows Defender not liking file writes.

Reply:
e.g. [`zip_fopen_index`](https://github.com/xamarin/LibZipSharp/blob/d321af1d2ac581bd0fd1e6a01f6c0b94ba1ee813/Native.cs#L123) can be used to open compressed data and copy it around as needed (you need to pass `OperationFlags.Compressed` to do that, here are the [upstream docs](https://libzip.org/documentation/zip_fopen_index.html))

Reply:
which is only publicly exposed via [`ZipEntry.Extract()`](https://github.com/xamarin/LibZipSharp/blob/0b1488b73e7e6ab5047f9831c316b98e0c862621/ZipEntry.cs#L222), so there's no *public* way to do this at present.

Reply:
It was never needed, so there was no reason to make it public (or, rather, wrapped in a method to call it)